File: Easy/appello/meta_appellodefaultview/meta_appellodefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_appellodefaultview
{
    public class Meta_appellodefaultview : Meta_easydata 
	{
        public Meta_appellodefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "appellodefaultview") {
				Name = "Appelli";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idappello"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "description", "Descrizione", nPos++);
						DescribeAColumn(T, "appello_aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "appelloazionekind_title", "Ordinario/Correttivo/Integrativo", nPos++);
						DescribeAColumn(T, "appellokind_title", "Tipologia", nPos++);
						DescribeAColumn(T, "sessione_idsessionekind", "Tipologia Sessione", nPos++);
						DescribeAColumn(T, "sessione_start", "Data di inizio Sessione", nPos++);
						DescribeAColumn(T, "sessione_stop", "Data di fine Sessione", nPos++);
						DescribeAColumn(T, "appello_minvoto", "Voto minimo", nPos++);
						DescribeAColumn(T, "appello_basevoto", "Votazione di base", nPos++);
						DescribeAColumn(T, "appello_prointermedia", "Prova intermedia", nPos++);
						DescribeAColumn(T, "appello_posti", "Numero massimo di posti", nPos++);
						DescribeAColumn(T, "appello_prenotstart", "Data di inizio prenotazioni", nPos++);
						if (T.Columns.Contains("appello_prenotstart")) T.Columns["appello_prenotstart"].ExtendedProperties["format"] = "g";
						DescribeAColumn(T, "appello_penotend", "Data fine delle prenotazioni", nPos++);
						if (T.Columns.Contains("appello_penotend")) T.Columns["appello_penotend"].ExtendedProperties["format"] = "g";
						DescribeAColumn(T, "appello_publicato", "Publicato", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/appello/meta_appellodefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_appellodefaultview() {
        MetaData.apply(this, ["appellodefaultview"]);
        this.name = 'meta_appellodefaultview';
    }

    meta_appellodefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_appellodefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'description', 'Descrizione', null, 1000, 1024);
						this.describeAColumn(table, 'aa', 'Anno accademico', null, 2000, 9);
						this.describeAColumn(table, 'appelloazionekind_title', 'Ordinario/Correttivo/Integrativo', null, 8200, 50);
						this.describeAColumn(table, 'appellokind_title', 'Tipologia', null, 9200, 50);
						this.describeAColumn(table, 'sessionekind_title', 'Tipologia Tipologia Sessione', null, 10320, 50);
						this.describeAColumn(table, 'sessione_start', 'Data di inizio Sessione', null, 10400, null);
						this.describeAColumn(table, 'sessione_stop', 'Data di fine Sessione', null, 10500, null);
						this.describeAColumn(table, 'appello_minvoto', 'Voto minimo', null, 14000, null);
						this.describeAColumn(table, 'appello_basevoto', 'Votazione di base', null, 15000, null);
						this.describeAColumn(table, 'appello_prointermedia', 'Prova intermedia', null, 16000, null);
						this.describeAColumn(table, 'appello_posti', 'Numero massimo di posti', null, 17000, null);
						this.describeAColumn(table, 'appello_prenotstart', 'Data di inizio prenotazioni', 'g', 18000, null);
						this.describeAColumn(table, 'appello_penotend', 'Data fine delle prenotazioni', 'g', 19000, null);
						this.describeAColumn(table, 'appello_publicato', 'Publicato', null, 20000, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idappello"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('appellodefaultview', new meta_appellodefaultview('appellodefaultview'));

	}());
